packages/did-test-utils/src/index.ts

```diff
@@ -0,0 +1,227 @@
+      signature: 'cccc',
+    },
+  ],
+}
+
+export const COMMITS = {
+  genesis: {
+    header: {
+      controllers: [DID_ID],
+    },
+    data: { much: 'data' },
+  },
+  genesisGenerated: {
+    jws: {
+      payload: 'bbbb',
+      signatures: [
+        {
+          protected:
+            'eyJraWQiOiJkaWQ6MzprMnQ2d3lmc3U0cGcwdDJuNGo4bXMzczMzeHNncWpodHRvMDRtdnE4dzVhMnY1eG80OGlkeXozOGw3eWRraT92ZXJzaW9uPTAjc2lnbmluZyIsImFsZyI6IkVTMjU2SyJ9',
+          signature: 'cccc',
+        },
+      ],
+      link: CID.parse('bafyreiau5pqllna6pewhp3w2hbvohxxeqsmffnwf6o2fwoln4ubbc6fldq'),
+    },
+    linkedBlock: {
+      data: {
+        much: 'data',
+      },
+      header: {
+        controllers: [DID_ID],
+      },
+    },
+  },
+  r1: {
+    desiredContent: { much: 'data', very: 'content' },
+    commit: {
+      jws: {
+        payload: 'bbbb',
+        signatures: [
+          {
+            protected:
+              'eyJraWQiOiJkaWQ6MzprMnQ2d3lmc3U0cGcwdDJuNGo4bXMzczMzeHNncWpodHRvMDRtdnE4dzVhMnY1eG80OGlkeXozOGw3eWRraT92ZXJzaW9uPTAjc2lnbmluZyIsImFsZyI6IkVTMjU2SyJ9',
+            signature: 'cccc',
+          },
+        ],
+        link: 'bafyreia6chsgnfihmdrl2d36llrfevc6xgmgzryi3ittdg3j5ohdifb7he',
+      },
+      linkedPayload: {
+        id: 'bafybeig6xv5nwphfmvcnektpnojts33jqcuam7bmye2pb54adnrtccjlsu',
+        data: [
+          {
+            op: 'add',
+            path: '/very',
+            value: 'content',
+          },
+        ],
+        prev: 'bafybeig6xv5nwphfmvcnektpnojts33jqcuam7bmye2pb54adnrtccjlsu',
+        header: {},
+      },
+    },
+  },
+  r2: { commit: { proof: FAKE_CID_4, id: FAKE_CID_1, prev: FAKE_CID_2 } },
+  proof: {
+    chainId: 'fakechain:123',
+  },
+}
+
+function resolutionResult(didUrl: string, optRotateDate?: string): DIDResolutionResult {
+  const rotateDate = optRotateDate || new Date('2022-03-11T21:28:07.383Z').toISOString()
+  const { did } = parseDidUrl(didUrl)!
+  const isVersion0 = /version=0/.exec(didUrl)
+
+  if (isVersion0) {
+    return {
+      didResolutionMetadata: { contentType: 'application/did+json' },
+      didDocument: wrapDocument(
+        {
+          publicKeys: {
+            signing: 'zQ3shwsCgFanBax6UiaLu1oGvM7vhuqoW88VBUiUTCeHbTeTV',
+            encryption: 'z6LSfQabSbJzX8WAm1qdQcHCHTzVv8a2u6F7kmzdodfvUCo9',
+          },
+        },
+        did
+      ),
+      didDocumentMetadata: {
+        nextUpdate: rotateDate,
+      },
+    }
+  }
+
+  return {
+    didResolutionMetadata: { contentType: 'application/did+json' },
+    didDocument: wrapDocument(
+      {
+        publicKeys: {
+          signing: 'zQ3shwsCgFanBax6UiaLu1oGvM7vhuqoW88VBUiUTCeHbTeTV',
+          encryption: 'z6MkjKeH8SgVAYCvTBoyxx7uRJFGM2a9HUeFwfJfd6ctuA3X',
+        },
+      },
+      did
+    ),
+    didDocumentMetadata: {
+      updated: rotateDate,
+    },
+  }
+}
+
+export class DidTestUtils {
```

Comment:
is this used for anything other than 3id-specific testing?  Can this be moved into the `3id-did-resolver` package?

Reply:
It's used for 3id, tile, and stream model testing.